pkg/bloomgateway/processor.go

```diff
@@ -113,16 +114,6 @@ func (p *processor) processTasks(ctx context.Context, tenant string, day config.
 }
 
 func (p *processor) processBlocks(ctx context.Context, bqs []*bloomshipper.CloseableBlockQuerier, data []blockWithTasks) error {
-
-	defer func() {
-		for i := range bqs {
-			if bqs[i] == nil {
-				continue
-			}
-			bqs[i].Close()
-		}
-	}()
```

Comment:
Maybe I miss something, but we don't seem to close the block remaining block queriers any more in case one `p.processBlock()` fails, since `concurrency.ForEachJob()` returns on first error.